allensdk/brain_observatory/behavior/vbo_2023_nwb_writers/behavior/__main__.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+import sys
+from pathlib import Path
+from typing import List
+
+import argschema
+import marshmallow
+from allensdk.brain_observatory.argschema_utilities import (
+    write_or_print_outputs,
+)
+from allensdk.brain_observatory.behavior.behavior_session import (
+    BehaviorSession,
+)
+from allensdk.brain_observatory.behavior.vbo_2023_nwb_writers.behavior.schemas import (  # noqa: E501
+    BehaviorInputSchema,
+    OutputSchema,
+)
+from allensdk.brain_observatory.behavior.vbo_2023_nwb_writers.nwb_writer_utils import (  # noqa: E501
+    _compare_metadata,
+)
+from allensdk.brain_observatory.session_api_utils import sessions_are_equal
+from pynwb import NWBHDF5IO
+
+
+def write_behavior_nwb(
+    behavior_session_id: int,
+    behavior_session_metadata: dict,
+    nwb_filepath: Path,
+    skip_metadata: List[str],
+    skip_stim: List[str],
+) -> dict:
+    """Load and write a BehaviorSession as NWB.
+
+    Check data object against associated data from the behavior_session
+    metadata table. Additionally, round trip the NWB file to confirm that it
+    was properly created for the session.
+
+    Parameters
+    ----------
+    behavior_session_id : int
+        Behavior session unique identifier.
+    behavior_session_metadata : dict
+        Dictionary of keys that overlap between the metadata in the
+        BehaviorSession object and the associated metadata table.
+    nwb_filepath : pathlib.Path
+        Base filename path to write the NWB to.
+    skip_metadata : list of str
+        List of metadata keys to skip when comparing data.
+    skip_stim : list of str
+        List of stimulus file keys to skip when comparing data.
+
+    Returns
+    -------
+    output_path : str
+        String path of where the NWB was written to.
+
+    Note
+    ----
+    Upon encountering an error, either from write or data comparison, the
+    code changes the file name suffix to error and does not return the
+    path of the file.
+    """
+
+    nwb_filepath_inprogress = str(nwb_filepath) + ".inprogress"
+    nwb_filepath_error = str(nwb_filepath) + ".error"
+
+    # Clean out files from previous runs:
+    for filename in [
+        nwb_filepath_inprogress,
+        nwb_filepath_error,
+        nwb_filepath,
+    ]:
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    try:
+        lims_session = BehaviorSession.from_lims(behavior_session_id)
+        nwbfile = lims_session.to_nwb()
+        with NWBHDF5IO(nwb_filepath_inprogress, "w") as nwb_file_writer:
+            nwb_file_writer.write(nwbfile)
+
+        logging.info(
+            "Comparing a BehaviorSession created from LIMS "
+            "to data from the behavior session metadata and the "
```

Comment:
I thought we were no longer performing validation of LIMS to the pickle files from the changes I made with removing from_json?

Reply:
Hey Mike, this is slightly different. When we release data we produce pandas/csv tables of metadata for the experiments/sessions. Until now, we haven't really validated the values in the metadata tables against those in the NWB file. There were more than a few conflicts initially so these updated functions are to catch those.